var/spack/repos/builtin/packages/r-knitr/package.py

```diff
@@ -31,17 +32,19 @@ class RKnitr(RPackage):
     depends_on("r@3.1.0:", type=("build", "run"), when="@1.15:1.22")
     depends_on("r@3.2.3:", type=("build", "run"), when="@1.23:")
     depends_on("r@3.3.0:", type=("build", "run"), when="@1.39:")
+
     depends_on("r-evaluate@0.10:", type=("build", "run"))
     depends_on("r-evaluate@0.15:", type=("build", "run"), when="@1.39:")
     depends_on("r-highr", type=("build", "run"))
-    depends_on("r-stringr@0.6:", type=("build", "run"))
+    depends_on("r-stringr@0.6:", type=("build", "run"), when="@:1.40")
```

Comment:
This is another nit-picky thing but can you move this no longer active dependency to the end of the file? That keeps the inactive dependencies grouped together.